RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -382,7 +376,7 @@ int PulseShapeFitOOTPileupCorrection::pulseShapeFit(const double * energyArr, co
       tmpslew[i] = 0;
       if(applyTimeSlew_) tmpslew[i] = HcalTimeSlew::delay(std::max(1.0,chargeArr[i]),slewFlavor_); 
       // add the noise components
-      tmperry2[i]=noise_*noise_+ noiseADCArr[i]*noiseADCArr[i];
+      tmperry2[i]=noiseArr[i]*noiseArr[i];
```

Comment:
it looks like noiseArr argument in pulseShapeFit is used only squared.
Also, it looks like the noiseArr derivation in the apply method is first computed as a square of the noise and sqrt is taken for no particular reason.
I suggest to pass noiseArr2 and eliminate the "sqr(sqrt" cycle and not waste CPU.
I guess this is also the origin of the small discrepancies in energy in pre-phase-1 workflows.